crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -147,6 +161,310 @@ async fn generate_cac(
+            let mut temp_c2 = c2.clone();
+            temp_c2.remove("req_payload");
+            let c2_val = temp_c2.remove("key_val");
+            if index2 != index1 {
+                if all_subsets_c1.contains(&temp_c2) {
+                    if c1_val == c2_val {
+                        can_be_removed = true;
+                        break;
+                    } else if c2_val.is_some() {
+                        break;
+                    }
+                }
+            }
+        }
+        if can_be_removed {
+            if let Some(req_payload) = temp_c1.get("req_payload") {
+                let mut t = Map::new();
+                t.insert("req_payload".to_string(), req_payload.clone());
+                temp_c1 = t;
+            };
+        }
+        dimensions[index1] = temp_c1;
+    }
+    dimensions
+}
+
+fn get_contextids_from_overrideid(
+    contexts: Vec<Context>,
+    overrides: Map<String, Value>,
+    key_val: Value,
+    override_id: &str,
+) -> Vec<(Context, Map<String, Value>, Value, String)> {
+    let mut res: Vec<(Context, Map<String, Value>, Value, String)> = Vec::new();
+    for ct in contexts {
+        let ct_dimensions = extract_dimensions(&ct.condition).unwrap();
+        if ct_dimensions.contains_key("variantIds") {
+            continue;
+        }
+        let override_keys = &ct.override_with_keys;
+        if override_keys.contains(&override_id.to_owned()) {
+            res.push((
+                ct,
+                overrides.clone(),
+                key_val.clone(),
+                override_id.to_string(),
+            ));
+        }
+    }
+    res
+}
+
+fn construct_new_payload(req_payload: &Map<String, Value>) -> web::Json<PutReq> {
+    let mut res = req_payload.clone();
+    res.remove("to_be_deleted");
+    res.remove("override_id");
+    res.remove("id");
+    if let Some(Value::Object(res_context)) = res.get("context") {
+        if let Some(Value::Object(res_override)) = res.get("override") {
+            return web::Json(PutReq {
+                context: res_context.to_owned(),
+                r#override: res_override.to_owned(),
+            });
+        }
+    }
+    web::Json(PutReq {
+        context: Map::new(),
+        r#override: Map::new(),
+    })
+}
+
+async fn reduce_context_key(
+    user: User,
+    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    mut og_contexts: Vec<Context>,
+    mut og_overrides: Map<String, Value>,
+    check_key: &str,
+    dimension_schema_map: &HashMap<String, (JSONSchema, i32)>,
+    default_config: Map<String, Value>,
+    is_approve: bool,
+) -> Config {
+    let default_config_val = default_config.get(check_key);
+    let mut contexts_overrides_values: Vec<(Context, Map<String, Value>, Value, String)> =
+        Vec::new();
+
+    for (key, val) in og_overrides.clone() {
+        match val {
+            Value::Object(obj_val) => {
+                if let Some(ans_val) = obj_val.get(check_key) {
```

Comment:
ans_val, obj_val need better names. The code is hard to understand

Reply:
If pass the value instead of fetching it from the default config, then no need if this `if let` block.

Reply:
This value if not coming from default config, it's coming from overrides, have updated the code with better names, please review